pos_barcode_tare/static/src/js/pos_barcode_tare.js

```diff
@@ -0,0 +1,273 @@
+        },
+    });
+
+    screens.define_action_button({
+        'name': 'tareScreenButton',
+        'widget': TareScreenButton,
+    });
+
+    // This is a new screen that reads weight from the electronic scale and
+    // create a barcode label encoding the weight. The screen shows a preview
+    // of the label. The user is expected to check if the preview matches what's
+    // measured on the scale. The barcode image is generated by the report
+    // module.
+    var TareScreenWidget = screens.ScreenWidget.extend({
+        template: 'TareScreenWidget',
+        next_screen: 'products',
+        previous_screen: 'products',
+        default_tare_value: 0.0,
+        weight_barcode_prefix: null,
+
+        show: function () {
+            this._super();
+            // Fetch the unit of measure used to save the tare
+            this.kg_unit = get_unit(this.pos, "kg");
+            // Fetch the barcode prefix from POS barcode parser rules.
+            this.weight_barcode_prefix = this.get_barcode_prefix();
+            // Setup the proxy
+            var queue = this.pos.proxy_queue;
+            // The pooling of the scale starts here.
+            var self = this;
+            queue.schedule(function () {
+                return self.pos.proxy.scale_read().then(function (weight) {
+                    self.set_weight(weight);
+                });
+            }, {duration:150, repeat: true});
+            // Shows a barcode whose weight might be zero, but this is preferred
+            // for UI/UX reasons.
+            this.render_receipt();
+            this.lock_screen(true);
+        },
+        get_barcode_prefix: function () {
+            var barcode_pattern = this.get_barcode_pattern();
+            return barcode_pattern.substr(0, 2);
+        },
+        get_barcode_pattern: function () {
+            var rules = this.get_barcode_rules();
+            var rule = rules.filter(
+                function (r) {
+                    // We select the first (smallest sequence ID) barcode rule
+                    // with the expected type.
+                    return r.type === tare_barcode_type;
+                })[0];
+            return rule.pattern;
+        },
+        get_barcode_rules: function () {
+            return this.pos.barcode_reader.barcode_parser.nomenclature.rules;
+        },
+        set_weight: function (scale_measure) {
+            var weight = scale_measure.weight;
+            var unit = get_unit(this.pos, scale_measure.unit);
+            if (weight > 0) {
+                this.weight_in_kg = convert_mass(weight, unit, this.kg_unit);
+                this.render_receipt();
+                this.lock_screen(false);
+            }
+        },
+        get_weight: function () {
+            if (typeof this.weight_in_kg === 'undefined') {
+                return this.default_tare_value;
+            }
+            return this.weight_in_kg;
+        },
+        ean13_checksum: function (s) {
```

Comment:
In Odoo core, there are some functions that could to be used, instead of rewriting it. For exemple, in barcodes modules : 
https://github.com/odoo/odoo/blob/13.0/addons/barcodes/static/src/js/barcode_parser.js#L54
odoo ``ean_checksum`` could replace your ``ean13_checksum``. Don't you think ? 
(I'm not sure however, because it's not the same algorithm.)

Reply:
Good idea, my function looks redundant. I'll fix this.